core/src/main/scala/cats/std/iterable.scala

```diff
@@ -0,0 +1,18 @@
+package cats
+package std
+
+trait IterableInstances {
+  implicit val iterableInstance: Foldable[Iterable] =
```

Comment:
Does this belong in cats core? In a recent [gitter conversation](https://gitter.im/non/cats?at=5633b2d0e4bb7eee537ffb51), the consensus was to put `IndexedSeq` instances in alleycats because `IndexedSeq` can potentially be mutable. Wouldn't the same arguments apply to `Iterable`?